bson/codec_options.py

```diff
@@ -294,26 +286,34 @@ class CodecOptions(_options_base):
 
     def __new__(
         cls: Type["CodecOptions"],
-        document_class: _DocumentClass = dict,
+        document_class: Optional[Type[Mapping[str, Any]]] = None,
         tz_aware: bool = False,
         uuid_representation: Optional[int] = UuidRepresentation.UNSPECIFIED,
         unicode_decode_error_handler: Optional[str] = "strict",
         tzinfo: Optional[datetime.tzinfo] = None,
         type_registry: Optional[TypeRegistry] = None,
     ) -> "CodecOptions":
-        if not (issubclass(document_class, _MutableMapping) or _raw_document_class(document_class)):
+        doc_class = document_class or dict
+        # issubclass can raise TypeError for generic aliases like SON[str, Any].
+        # In that case we can use the base class for the comparison.
+        is_mapping = False
+        try:
+            is_mapping = issubclass(doc_class, _MutableMapping)
+        except TypeError:
+            is_mapping = issubclass(doc_class.mro()[0], _MutableMapping)
+        if not (is_mapping or _raw_document_class(doc_class)):
             raise TypeError(
                 "document_class must be dict, bson.son.SON, "
                 "bson.raw_bson.RawBSONDocument, or a "
-                "sublass of collections.abc.MutableMapping"
+                "subclass of collections.abc.MutableMapping"
             )
         if not isinstance(tz_aware, bool):
             raise TypeError("tz_aware must be True or False")
         if uuid_representation not in ALL_UUID_REPRESENTATIONS:
             raise ValueError(
                 "uuid_representation must be a value from bson.binary.UuidRepresentation"
             )
-        if not isinstance(unicode_decode_error_handler, (str, None)):  # type: ignore
+        if not isinstance(unicode_decode_error_handler, (str, None)):  # type: ignore[arg-type]
```

Comment:
I think this is hiding a real bug:
```
>>> isinstance(None, (str, None))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
```

Let's remove None here and make unicode_decode_error_handler non-Optional since None was never supported.

Reply:
Done